### PR TITLE
🔒 enable security settings for the main window

### PR DESCRIPTION
**Pull request by MarshallOfSound**
These include `contextIsolation: true`, `sandbox: true`, `nodeIntegration: false` and `enableRemoteModule: false`

Refs: #343 #342 #341

Was actually surprisingly straight forward, most of the code worked out of the box. Not sure if I missed anything though so would be great if someone more familiar with the codebase gave it a whirl.

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -239,9 +239,12 @@ function createWindow() {
 
     view = new BrowserView({
         webPreferences: {
-            nodeIntegration: true,
-            webviewTag: true,
-            enableRemoteModule: true,
+            nodeIntegration: false,
```
Comment (adlerluiz) [inline main.js:242]: When performing the test with these modifications, the buttons injected into the interface no longer appear.

image

Comment (MarshallOfSound) [reply main.js:242]: Hm they appeared on my local, let me double check.  We're there any errors in your devtools for the BrowserView?

Comment (adlerluiz) [reply main.js:242]: image

**Comment by MarshallOfSound** [comment]
@adlerluiz Can you confirm you're on the latest version of my branch.  I just double checked and I definitely get all the injected UI 🤔  That errors look like you're not using my version of `injectControls.js` because I completely removed that `require` call 🤔

**Comment by adlerluiz** [comment]
I cloned in another directory and now works correctly 😊

File: src/providers/settingsProvider.js [hunk 2]
```diff
@@ -30,8 +31,45 @@ function setInitialValue(settingName, initialValue) {
 }
 
 function onDidChange(key, callback) {
-    store.onDidChange(key, (newValue, oldValue) => {
-        callback({ newValue: newValue, oldValue: oldValue })
+    return store.onDidChange(key, (newValue, oldValue) => {
+        callback({ newValue, oldValue })
```
Comment (adlerluiz) [inline src/providers/settingsProvider.js:35]: image

changing to callback(newValue, oldValue)

image

Comment (MarshallOfSound) [reply src/providers/settingsProvider.js:35]: Fixed 😄

**Pull request merged**